src/init.sh

```diff
@@ -170,5 +181,179 @@ function init_help()
     '  init --remote <user>@<ip>:<port> - Set remote fields in the kworkflow.config file.' \
-    '  init --target <target> Set the default_deploy_target field in the kworkflow.config file'
+    '  init --target <target> - Set the default_deploy_target field in the kworkflow.config file' \
+    '  init --guided-setup - Perform first time setup in an interactive manner. Recommended for new users.'
+}
+
+# This function gets four configuration values from git: name, email,
+# editor, and initial branch name and stores them inside an associative array
+# for future access. Checks if these values are empty and store this information
+# in a boolean variable inside the array.
+#
+# @_git_config: associative array received as a nameref attribute to store all
+#              values acquired in this function.
+#
+# Returns: _git_config[configured]: 1, if there is any revelant information in
+#          git missing, and 0, if git is fully configured.
+function get_git_config()
+{
+  local -n _git_config="$1"
+
+  _git_config['name']=$(git config user.name)
+  _git_config['email']=$(git config user.email)
+  _git_config['editor']=$(git config core.editor)
+  _git_config['branch']=$(git config init.defaultBranch)
+
+  _git_config['configured']=0
+
+  if [[ -z "${_git_config['name']}" ||
+    -z "${_git_config['email']}" ||
+    -z "${_git_config['editor']}" ||
+    -z "${_git_config['branch']}" ]]; then
+    _git_config['configured']=1
+  fi
+}
+
+# This function sets if the scope of the git configuration will be
+# local or global.
+#
+# @config_cmd: command in which will be added the configuration scope.
+#
+# Returns: the git config command, adjusted if the scope is local or global
+function set_git_config_scope()
+{
+  local command
+  local scope
+
+  command='git config'
+
+  if ! is_inside_work_tree; then
+    warning 'You are not in a git repository. All modifications will be global' 1>&2
+    if [[ $(ask_yN 'Would you like to proceed?' 'y') =~ '1' ]]; then
+      command+=' --global'
+    else
+      command=''
+    fi
+
+    printf '%s\n' "$command"
+    exit 0
+  fi
+
+  printf '\nSelect the scope of this configuration:\n' 1>&2
+
+  select scope in 'Local' 'Global'; do
+    case "$scope" in
+      'Global')
+        command+=' --global'
+        break
+        ;;
+      'Local')
+        command+=' --local'
+        break
+        ;;
+    esac
+  done
+
+  printf '%s\n' "$command"
+}
+
+# Allows the user to set the empty configurations if any.
+#
+# @_config_git: associative array with data about git configuration.
+# @flag: flag sended to the cmd_manager to change its behaviour.
+#
+# Returns: _config_git: array with the selected information configured on git
+function interactive_set_user_git_info()
+{
+  local -n _config_git="$1"
+  local flag="$2"
+  local git_editor_suggestion
+  local config_cmd
+  local scope
+
+  flag=${flag:-'SILENT'}
+
+  if [[ "${_config_git['configured']}" == '0' ]]; then
+    exit 0
```

Comment:
Let use return. Also, I think we should tell the user that git is already configured and maybe show the current config.